src/plugins/adonis.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const web = require('./util/web')
+
+function createWrapMethod (tracer, config) {
+  config = web.normalizeConfig(config)
+
+  return function wrapMethodRequest (handle) {
+    return function methodWithTrace (request, response) {
+      const route = this.Route.match(request.url, request.method, request.hostname)
+      return web.instrument(tracer, config, request, response, 'adonis.request', () => {
+        const span = web.active(request)
+
+        if (span) {
+          span.setTag('component', 'adonis')
+        }
+        if (!route) {
+          route.route.name = '404'
+        }
+        web.exitRoute(request)
+        web.enterRoute(request, route.route.name)
+        return handle.apply(this, arguments)
+      })
+    }
+  }
+}
+
+function createWrapResolve (tracer, config) {
+  return function wrapResolve (_resolveMiddleware) {
+    return function resolveWithTrace (middleware, ctx) {
+      if (typeof middleware.namespace === 'string') {
+        middleware.name = middleware.namespace.split('/').slice(-1)[0].split('.')[0]
+      } else if (typeof middleware.namespace === 'function') {
+        if (typeof middleware.params[0] === 'string') {
+          middleware.name = middleware.params['0']
+        } else {
+          middleware.name = 'adonis.middleware'
+        }
+      }
+      const request = ctx[0].req
```

Comment:
looks like `ctx` is actually an arguments list, so renaming seems helpful.  Does this work with all supported signatures?

Reply:
It has worked with my tests so far, but I can't say how exhaustive they are